site/content/blog/embeddings-database-news-april-2023/index.md

```diff
@@ -0,0 +1,361 @@
+  - These are just helpers though
+    and you can always just write your own functions to do the same thing.
+
+The term "vector" itself comes from linear algebra, which is a branch of mathematics that deals primarily with matrices,
+tensors, and other array-like structures, but at the end of the day, they're just fancy float arrays!
+
+That does beg the question though: If they're just float arrays, and since databases can already store arrays, do we
+*really* need a whole new database just for them? (Are VCs trying to start another bubble like they did with NoSQL
+databases?)
+
+### Why are Vector Databases necessary?
+
+Fortunately, the answer is simple: **Performance**.
+
+Vector Databases are optimized for storing and performing operations on large amounts of vector data, often processing
+hundreds of millions of vectors per query, and doing it significantly faster than traditional databases are able to.
+
+Here are some of the features that Vector Databases provide:
+- Perform complex mathematical operations to filter and locate "nearby" vectors using clustering techniques like
+  "Cosine Similarity" (which I'll explain later)
+- Provide [specialized Vector indexes](https://weaviate.io/developers/weaviate/more-resources/performance) to make retrieval
+  of data significantly faster and deterministic
+- Store vectors in a way that makes them more compact, like by [compressing and quantizing them](https://weaviate.io/blog/ann-algorithms-hnsw-pq),
+  to keep as much data in memory as possible (further reducing load + query latency)
+- Sharding data across multiple machines to handle large amounts of data (which many databases can do, but
+  SQL databases in particular take more effort to scale out)
+
+### Relational vs Document vs Vector Databases
+
+#### Relational Databases (SQL)
+
+Examples include:
+PostgreSQL, MySQL, and MariaDB.
+
+They use rows and columns to store  data, similar to an Excel spreadsheet, and they support Vectors with extensions like
+[pgvector](https://github.com/pgvector/pgvector). They're not as trivially scalable as a dedicated Vector Database, but
+they're much easier to deploy and manage due to their popularity with services like AWS RDS letting you spin up a
+production box in a few minutes.
+
+#### Document Databases (often "NoSQL")
+
+MongoDB and CouchDB are the quintessential examples here, but technically Redis also sits in this space.
+
+They tend to store data in JSON-like structures, called "documents", that can are easy to work with, but they aren't
+being used in the AI world very much today (most likely because they're not as popular as SQL databases today). The
+exception is Redis which does have
+[Vector support](https://redis.io/docs/stack/search/reference/vectors/) and is quite popular for in-memory use cases.
+
+#### Vector Databases
+
+These databases are very "niche" in the problem they solve. They are not general databases that you can use to build the
```

Comment:
Mention FAISS, a lot of these vector database are just hosted instances of this. Also include this as further reading https://www.pinecone.io/learn/faiss/